draft-ietf-tls-esni.md

```diff
@@ -347,14 +353,48 @@ For servers (in EncryptedExtensions), this extension contains the following
 structure:
 
 ~~~
+   enum {
+       esni_accept(0),
+       esni_retry_request(1),
+   } ServerESNIResponseType;
+
    struct {
-       uint8 nonce[16];
+       ServerESNIResponseType response_type;
+       select (response_type) {
+           case esni_accept:        uint8 nonce[16];
+           case esni_retry_request: ESNIKeys retry_keys<1..2^16-1>;
```

Comment:
We will evolve the format over time.  That means that this field is going to contain a specific version.  If that version is not supported, this design offers no means of recovery.  The client knows for certain that ESNI is supported, but there is no remedy.

Can this include a list of key structures?  That would be consistent with the DNS, which can have multiple alternative records with different versions.

Reply:
I see.  It's not very clear.  I'll make some suggestions.